test/unit/govuk_index/elasticsearch_processor_test.rb

```diff
@@ -20,4 +20,31 @@ def test_should_save_valid_document
     actions.save(presenter)
     actions.commit
   end
+
+  def test_should_delete_valid_document
+    presenter = stub(:presenter)
+    presenter.stubs(:identifier).returns(
+      _type: "cheddar",
+      _id: "/cheese"
+    )
+    presenter.stubs(:document).returns(
+      link: "/cheese",
+      title: "We love cheese"
+    )
+
+    client = stub('client')
+    Services.stubs('elasticsearch').returns(client)
+    client.expects(:bulk).with(
+      index: SearchConfig.instance.govuk_index_name,
+      body: [
+        {
+          delete: presenter.identifier
+        }
+      ]
+    )
+
+    actions = GovukIndex::ElasticsearchProcessor.new
+    actions.delete(presenter)
+    actions.commit
+  end
```

Comment:
We currently raise/rescue the `UnknownDocumentTypeError` from the `DocumentTypeInferer` so it shouldn't ever try to delete a document if the payload document type is invalide